documentation/source/ptq_qat.md

```diff
@@ -0,0 +1,233 @@
+
+```python
+from super_gradients.training.utils.quantization.selective_quantization_utils import register_quantized_module
+from super_gradients.training.utils.quantization.selective_quantization_utils import QuantizedMetadata
+
+@register_quantized_module(float_source=MyNonQuantBlock,
+                           action=QuantizedMetadata.ReplacementAction.REPLACE,
+                           input_quant_descriptor=QuantDescriptor(...),
+                           weights_quant_descriptor=QuantDescriptor(...)
+                           )
+class MyQuantBlock:
+    ...
+```
+### Quantizing residuals and skip connections
+To improve performance of quantized models, quantization of residuals and skip connections is performed. SuperGradients API allow you to do it. In your source code, add one of the following, depending on the type of the skip connection: 
+
+```python
+from super_gradients.modules.skip_connections import (
+    Residual, 
+    SkipConnection, 
+    CrossModelSkipConnection, 
+    BackboneInternalSkipConnection, 
+    HeadInternalSkipConnection
+)
+```
+
+Use them for all inputs of the `sum`, `mul`, `div` and `concat` operations. `SelectiveQuantizer` will take care of them and will replace them with quantized counterparts. 
+
+### Calibration
+And after quantization, performing calibration take another two lines of code:
+
+```python
+from super_gradients.training.utils.quantization.calibrator import QuantizationCalibrator
+model = ... # your quantized model
+calib_dataloader = ... # your standard pytorch dataloader
+
+calibrator = QuantizationCalibrator(verbose=True)
+calibrator.calibrate_model(
+    model,
+    method="percentile",
+    calib_data_loader=calib_dataloader,
+    num_calib_batches=16,
+    percentile=99.99,
+)
+```
+
+Your model is now quantized and calibrated!
+
+Refer to `super_gradients/src/super_gradiens/examples/quantization` for more source examples that are ready-to-run!
+
+
+## Quantization-Aware training
+
+Quantization-aware training (QAT) is a method that takes into account the impact of quantization during the training process. The model is trained with quantization-aware operations that mimic the quantization process during training. This allows the model to learn how to perform well in the quantized representation, leading to improved accuracy compared to post-training quantization.
+
+With SuperGradients, after you have done PTQ, you can finetune your quantized model with standard training pipeline:
+
+```python
+from super_gradients import Trainer
+
+model = ...  # your quantized and calibrated model
+train_dataloader = ... # your standard pytorch dataloader
+valid_dataloader = ... # your standard pytorch dataloader
+training_hyperparams = ... # refer to training_hyperparams example to fill it
+
+model.train()
+trainer = Trainer(experiment_name="my_first_qat_experiment", ckpt_root_dir=...)
+res = trainer.train(
```

Comment:
Shouldn't it be `QATTrainer` or something

Reply:
No, QATTrainer implements only train_from_config
Regular Trainer with quantized model